python/lsst/drp/tasks/update_visit_summary.py

```diff
@@ -0,0 +1,755 @@
+    record: ExposureRecord, bbox: Box2I | None = None, wcs: SkyWcs | None = None
+) -> SpherePoint | None:
+    """Compute the sky coordinate center for a detector to be used when
+    testing distance to tract center.
+
+    Parameters
+    ----------
+    record : `lsst.afw.table.ExposureRecord`
+        Exposure record to obtain WCS and bbox from if not provided.
+    bbox : `lsst.geom.Box2I`, optional
+        Bounding box for the detector in its own pixel coordinates.
+    wcs : `lsst.afw.geom.SkyWcs`, optional
+        WCS that maps the detector's pixel coordinate system to celestial
+        coordinates.
+
+    Returns
+    -------
+    center : `lsst.geom.SpherePoint` or `None`
+        Center of the detector in sky coordinates, or `None` if no WCS was
+        given or present in the given record.
+    """
+    if bbox is None:
+        bbox = record.getBBox()
+    if wcs is None:
+        wcs = record.getWcs()
+        if wcs is None:
+            return None
+    region = makeSkyPolygonFromBBox(bbox, wcs)
+    return SpherePoint(region.getCentroid())
+
+
+class PossiblyMultipleInput(ABC):
+    """A helper ABC for handling input `~lsst.afw.table.ExposureCatalog`
+    datasets that may be multiple (one per tract/visit combination) or
+    unique/global (one per visit).
+    """
+
+    @abstractmethod
+    def best_for_detector(
+        self,
+        detector_id: int,
+        center: SpherePoint | None = None,
+        bbox: Box2I | None = None,
+    ) -> tuple[int, ExposureRecord | None]:
+        """Return the exposure record for this detector that is the best match
+        for this detector.
+
+        Parameters
+        ----------
+        detector_id : `int`
+            Detector ID; used to find the right row in the catalog or catalogs.
+        center : `SpherePoint` or `None`
+            Center of the detector in sky coordinates.  If not provided, one
+            will be computed via `compute_center_for_detector_record`.
+        bbox : `lsst.geom.Box2I`, optional
+            Bounding box for the detector in its own pixel coordinates.
+
+        Returns
+        -------
+        tract_id: `int`
+            ID of the tract that supplied this record, or `-1` if ``record`` is
+            `None` or if the input was not per-tract.
+        record : `lsst.afw.table.ExposureRecord` or `None`
+            Best record for this detector, or `None` if there either were no
+            records for this detector or no WCS available to compute a center.
+        """
+        raise NotImplementedError()
+
+
+@dataclasses.dataclass
+class PerTractInput(PossiblyMultipleInput):
+    """Wrapper class for input `~lsst.afw.table.ExposureCatalog` datasets
+    that are per-tract.
+    """
+
+    catalogs_by_tract: list[tuple[TractInfo, ExposureCatalog]]
+    """List of tuples of catalogs and the tracts they correspond to
+    (`list` [ `tuple` [ `lsst.skymap.TractInfo`,
+        `lsst.afw.table.ExposureCatalog` ] ] ).
```

Comment:
Mostly I add spaces out of paranoia whenever there's nesting, since I hadn't internalized Sphinx's rules for when it needs them.  Thanks for the tip in the later message.